docs/source/cpp/flight.rst

```diff
@@ -239,7 +239,16 @@ Memory management
 -----------------
 
 Flight tries to reuse allocations made by gRPC to avoid redundant
-data copies. However, this means that those allocations may not
+data copies. Experience shows that Arrow data in those allocations are not
+found at aligned addresses. Some Flight client use cases may require `data alignment`_,
```

Comment:
```suggestion
found at aligned addresses. Some Flight client use cases may require :ref:`data alignment<buffer-alignment-and-padding>`,
```

with

```diff
diff --git a/docs/source/format/Columnar.rst b/docs/source/format/Columnar.rst
index e1603e8d8e..5a7ca963ce 100644
--- a/docs/source/format/Columnar.rst
+++ b/docs/source/format/Columnar.rst
@@ -258,6 +258,8 @@ in whichever form is convenient for them. We handle metadata
 **serialization** in an implementation-independent way using
 `Flatbuffers`_, detailed below.
 
+.. _buffer-alignment-and-padding:
+
 Buffer Alignment and Padding
 ----------------------------
 
```

Reply:
No, I don't think this suggestion is correct @kou. What we're doing here is ensuring that is naturally aligned, e.g. int32 aligned to 4 bytes, not 64-byte aligned as recommended in the columnar spec.

I would recommend something like this:
```
Flight tries to reuse allocations made by gRPC to avoid redundant
data copies. However, experience shows that such data is frequently
misaligned. Some use cases might require data to be "naturally"
aligned (for example, for the data buffer of a Int32 array to be aligned
on a 4-byte boundary), which can be enforced ...
```

Reply:
I went for the term "data type-specific" alignment instead of "natural" here.
https://github.com/apache/arrow/blob/b3e607b6d53ba7d26791ba6c2b5bda64d9b0e709/cpp/src/arrow/util/align_util.h#L74-L81